app/assets/javascripts/account/AccountService.js.coffee

```diff
@@ -123,11 +124,12 @@ AccountService = (
         $state.go('dahlia.my-applications')
     ).catch (response) ->
       response = response.data if response.data
-      msg = response.errors.full_messages[0]
-      if msg == 'Current password is invalid'
-        msg = $translate.instant("error.current_password_invalid")
-      else
-        msg = $translate.instant("error.password_update")
+      if $window.env.ACCOUNT_INFORMATION_PAGES_REACT == "false"
+        msg = response.errors.full_messages[0]
+        if msg == 'Current password is invalid'
+          msg = $translate.instant("error.current_password_invalid")
+        else
+          msg = $translate.instant("error.password_update")
```

Comment:
suggestion: Re-use the fix from https://github.com/SFDigitalServices/sf-dahlia-web/pull/2344 for consistency (include the comment as well), and also to avoid the need to check for env vars. e.g.

```
    ).catch (response) ->
      if (error.message == 'transition prevented')
        ### TODO:
          If the user is navigating from an Angular Sign In page to a React my account page,
          the ui-router will prevent the transition and force routing via Rails.
          Preventing the transition causes the ui-router to throw an error, which we catch here.
          Setting the timeout allows the Rails routing system time to catch up and
          route the user to the React My Account page without throwing an error.
          This is a temporary solution until we can launch the re-written Sign In page.
        ###
        setTimeout (-> $scope.handleErrorState()), 1000
      else
        response = response.data if response.data
        msg = response.errors.full_messages[0]
        if msg == 'Current password is invalid'
          msg = $translate.instant("error.current_password_invalid")
        else
          msg = $translate.instant("error.password_update")
```

Reply:
Thanks for the explanation. I've linked the Jira tickets so it's more obvious that the PR I referenced caused issues.